specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2020-07-01-preview/examples/CreateExtension.json

```diff
@@ -0,0 +1,61 @@
+{
+  "parameters": {
+    "subscriptionId": "subId1",
+    "resourceGroupName": "rg1",
+    "clusterRp": "Microsoft.Kubernetes",
+    "clusterResourceName": "connectedClusters",
+    "extensionInstanceName": "ClusterMonitor",
+    "api-version": "2020-07-01-preview",
+    "clusterName": "clusterName1",
+    "extensionInstance": {
+      "properties": {
+        "extensionType": "azuremonitor-containers",
+        "autoUpgradeMinorVersion": true,
+        "releaseTrain": "Preview",
+        "scope": {
+          "cluster": {
+            "releaseNamespace": "kube-system"
+          }
+        },
+        "configurationSettings": {
+          "omsagent.secret.wsid": "a38cef99-5a89-52ed-b6db-22095c23664b",
+          "omsagent.env.clusterName": "clusterName1"
+        },
+        "configurationProtectedSettings": {
+          "omsagent.secret.key": "secretKeyValue01"
+        }
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "headers": {},
+      "description": "Details of the Kubernetes Extension's current status.",
+      "body": {
+        "id": "/subscriptions/subId1/resourceGroups/rg1/providers/Microsoft.Kubernetes/connectedClusters/clusterName1/providers/Microsoft.KubernetesConfiguration/extensions/ClusterMonitor",
+        "type": "Microsoft.KubernetesConfiguration/extensions",
+        "name": "ClusterMonitor",
+        "properties": {
+          "extensionType": "azuremonitor-containers",
+          "autoUpgradeMinorVersion": true,
+          "releaseTrain": "Preview",
+          "version": "0.1.4",
+          "scope": {
+            "cluster": {
+              "releaseNamespace": "kube-system"
+            }
+          },
+          "configurationSettings": {
+            "omsagent.secret.wsid": "a38cef99-5a89-52ed-b6db-22095c23664b",
+            "omsagent.env.clusterName": "clusterName1"
```

Comment:
we shouldn't expose this since this being auto populated by config rp

Reply:
ConfigurationSettings is a key/value pair that can take any key/value combination.  These key/values shown here are just examples and as such need not be provided by the user.